### PR TITLE
Add browser-accessible Dev-Frame

### PR DESCRIPTION
**Pull request by siefkenj**
Adds `dev-frame.html`, which is built when `dev` is built. In the dev frame you can test and rapid-develop Conversations components.

This PR s based on #1583. It is largely a proof-of-concept, but with files being watched and automatically recompiled, it makes development a lot faster! Also, there are clear errors when accessing Thunderbird-only APIs, so it should be easier to identify and split components, which should help in testing.

`MessageIFrame` seems to be deeply integrated with Thunderbird at the moment, so I inserted an `ErrorBoundary` component and just let it error for the time being.

### CODE REVIEW

**Comment by siefkenj** [comment]
For some reason, this PR is causing the `browser` object to use the fallback version even though a real version should be available. Any idea why?

**Comment by siefkenj** [comment]
@Standard8 Any ideas when you might be able to review this?

**Review by Standard8 — CHANGES_REQUESTED** [review]
This looks good to me, I do wonder if we should do this from the message list/conversation level, but it is a good start anyway.

I also wondered if we could do something like a console tracker for `browser.*` calls and have those output, for example you could click the reply button and check the call and its arguments.

We could do that later though, this is a good start in itself.

**Comment by Standard8** [comment]
And sorry for not responding earlier. I did think I had put some comments on it but obviously I hadn't got all the way through.

File: addon/content/modules/message.js [hunk 1]
```diff
@@ -2,8 +2,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-"use strict";
-
```
[the remark this answers is not in the document]
Comment (siefkenj) [reply addon/content/modules/message.js:6]: If something is imported as a module, doesn't it automatically get a `"use strict"`? In any case, I'll add it back.

**Comment by siefkenj** [comment]
> I also wondered if we could do something like a console tracker for `browser.*` calls and have those output, for example you could click the reply button and check the call and its arguments.

You're thinking of this only when running in the browser, or in general whenever debug messages are enabled?

File: addon/content/modules/message.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Standard8) [reply addon/content/modules/message.js]: Not for the legacy modules that Firefox uses - these aren't es6 modules (yet).

**Comment by Standard8** [comment]
> > I also wondered if we could do something like a console tracker for `browser.*` calls and have those output, for example you could click the reply button and check the call and its arguments.
> 
> You're thinking of this only when running in the browser, or in general whenever debug messages are enabled?

I was thinking of only when running in the browser - maybe a separate text box or something.

**Review by Standard8 — APPROVED** [review]

**Pull request merged**